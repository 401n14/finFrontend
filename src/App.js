import React from "react";
import Chat from "./components/Chat";

import "./styles/styles.scss";

export default function App() {
  return (
    <div className="App">
      <Chat />
    </div>
  );
}
