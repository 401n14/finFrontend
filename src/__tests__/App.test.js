import React from 'react';
import App from '../App';

import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import { useAuth0 } from '../react-auth0-spa.js';


configure({adapter:new Adapter()});

jest.mock('../react-auth0-spa.js');

describe('Snapshot Test', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            loading: false
        });
    });
    it('renders correctly logged in, Snapshot', () => {
        const page = renderer.create(
        <BrowserRouter>
        <App />
        </BrowserRouter>
        ).toJSON();
        expect(page).toMatchSnapshot();
    });
});
describe('Works when loading', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            loading: true
        });
    });
    it('renders correctly', () => {
            let wrapper = shallow(<App />)
            expect(wrapper.text()).toEqual('<Loading />');
    });
});