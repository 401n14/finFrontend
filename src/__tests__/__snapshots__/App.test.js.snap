// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test renders correctly logged in, Snapshot 1`] = `
Array [
  <div
    className="nav-container"
  >
    <nav
      className="navbar navbar-expand-md"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0)",
        }
      }
    >
      <div
        className="container"
      >
        <svg
          className="header-svg"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 4H5a1 1 0 1 1 0-2h11V1a1 1 0 0 0-1-1H4a2 2 0 0 0-2 2v16c0 1.1.9 2 2 2h12a2 2 0 0 0 2-2V5a1 1 0 0 0-1-1h-7v8l-2-2-2 2V4z"
          />
        </svg>
        <h1
          className="header-text"
        >
          Transcribe
        </h1>
        <button
          aria-label="Toggle navigation"
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="header-nav navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="header-nav-text nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="header-nav-text"
                href="#about"
                onClick={[Function]}
              >
                About Us
              </a>
            </li>
          </ul>
          <ul
            className="d-none d-md-block navbar-nav"
          >
            <li
              className="nav-item"
            >
              <button
                className="btn btn-outline-white btn-small"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Login
              </button>
            </li>
          </ul>
          <ul
            className="d-md-none navbar-nav"
          >
            <li
              className="nav-item"
            >
              <button
                className="btn btn-outline-white btn-small"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Login
              </button>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="contain"
  >
    <div
      className="col-2 mobile"
    >
      <div
        className="triple mobile"
      >
        <div
          className="triple-img"
          style={
            Object {
              "backgroundImage": "url(https://live.staticflickr.com/5813/30931763465_c7cc26e743_n.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="triple-img"
          style={
            Object {
              "backgroundImage": "url(https://live.staticflickr.com/8568/15486971303_fe3c25285d.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
        <div
          className="triple-img"
          style={
            Object {
              "backgroundImage": "url(https://live.staticflickr.com/1735/40811146810_d47a60d20c.jpg)",
              "backgroundSize": "cover",
            }
          }
        />
      </div>
    </div>
    <div
      className="col-2 hero-image"
    >
      <h1
        className="primary bold width"
      >
        Transcribe is
        <br />
        <span
          className="primary light"
        >
          your solution to modern communication
        </span>
      </h1>
      <p
        className="secondary "
      >
        The application is a communication platform for realtime chat between two or more users. We utilized Google Translate API to translate one user's preferred language to the other user's preferred language. Essentially, users can communicate easily and effectively in multiple languages at the same time.
      </p>
      <div
        className="hero-buttons"
      >
        <button
          className="btn btn-primary"
        >
          chat
        </button>
        <a
          href="https://github.com/401n14/finFrontend"
        >
          <button
            className="btn btn-outline-white"
          >
            Learn  more
          </button>
        </a>
      </div>
    </div>
  </div>,
  <hr />,
  <div
    className="next-steps my-5"
  >
    <h2
      className="my-5 text-center"
      id="about"
    >
      Learn more about us!
    </h2>
    <div
      className="d-flex justify-content-between row"
    >
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/james-ad"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            James Dunn - Lead Developer
          </a>
        </h6>
        <p>
          I’m a lifelong musician. I’ve played guitar and drums since I was a youngster. I received my M.A. in Philosophy from Brandeis University in Massachusetts
        </p>
      </div>
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/NadyaIlinskiy"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Nadya Ilinskaya - DevOps
          </a>
        </h6>
        <p>
          I’m breaking software for living, but really enjoy learning how to build it. 
        </p>
      </div>
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/all1124"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Anna Li - Project Manager
          </a>
        </h6>
        <p>
          I’m originally from Ohio, moved to Washington 3 years ago.
        </p>
      </div>
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/malemu17"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Meron Sibani - Full Stack Developer
          </a>
        </h6>
        <p>
            I am originally from Ethiopia moved to States three years ago. Currently I am raising my beautiful daughter
        </p>
      </div>
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/Martinbalke"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Martin Balke - Lead Designer
          </a>
        </h6>
        <p>
            I love solving problems. I’m a professional chef and baker currently. I love going to farmers markets and cooking with all of the great ingredients in Washington
        </p>
      </div>
      <div
        className="mb-4 col-md-5"
      >
        <h6
          className="mb-3"
        >
          <a
            href="https://github.com/mts513"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-link fa-w-16 mr-2"
              data-icon="link"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Morgan Shaw - Lead Tester 
          </a>
        </h6>
        <p>
           I am from Texas & graduated from Texas A&M University. I currently work for Zillow Group. I am a proud parent of a Saint Bernard & a black cat
        </p>
      </div>
    </div>
  </div>,
  <footer
    className="bg-light p-3 text-center"
  >
    <p
      style={
        Object {
          "marginTop": "2.5em",
        }
      }
    >
      <b>
        nightJAMMM
      </b>
       team © CodeFellows | 2020
    </p>
  </footer>,
]
`;
